tests/test_deployments/test_deployments.py

```diff
@@ -255,6 +261,34 @@ def test_creates_ropsten_deploy(self) -> None:
 class TestDeploymentGenerators(BaseDeploymentTests):
     """Test functionality of the deployment generators."""
 
+    def test_read_invalid_keys_file(self) -> None:
+        """Test JSONDecodeError on read_keys"""
+
+        side_effect = json.decoder.JSONDecodeError("", "", 0)
+        expected = "Error decoding keys file, please check the content of the file"
+        with mock.patch.object(json, "loads", side_effect=side_effect):
+            with pytest.raises(NotValidKeysFile, match=expected):
+                ServiceBuilder.read_keys(mock.Mock(), DEFAULT_KEY_PATH)
+
+    def test_update_agent_number_based_on_keys_file(self) -> None:
+        """Test JSONDecodeError on read_keys"""
+
+        public_id = PublicId("Georg", "Hegel")
+        service = Service(
+            "Arthur", "Schopenhauer", public_id, number_of_agents=1_000_000
+        )
```

Comment:
shall I open an issue on this on the open-aea side? Is there a check somewhere I don't know of that ensures public_ID author and name are the same as those on the components?